src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -4077,12 +4092,24 @@ new_readdir(DIR *dirp)
 	return &mydir->ents[mydir->num_ents];
 }
 
+static char env_str_int_on[]  = "D_IL_INTERCEPTION_ON=1";
+static char env_str_int_off[] = "D_IL_INTERCEPTION_ON=0";
```

Comment:
i do not get what this D_IL_INTERCEPTION_ON env is about and why we need it?
when was this added?

Reply:
You previously suggested that we should make whitelist mode as default. Users always run in whitelist mode.
I added "D_IL_INTERCEPTION_ON" as a new env in recent revision of this PR. This is mainly used for testing to make interception all the time. We do not expose this env to users in documentation.

Reply:
please add a comment here to explain clearly this case.
i still don't get why we need that for testing. when we test, we want to use the whitelist and bypass interception of those, no?
or are you saying you also want to add some tests to override the whitelist just to check those work even if performance is not good?
If the latter, the name of the env is very confusing. to me, if one set LD_PRELOAD that means interception is on with some exceptions. this sounds exactly the same, but the behavior is different. since it's test only, maybe just call it:
D_IL_NO_WHITELIST
making it more clear?

Reply:
ok. I will add comments as you suggested.

Yes. We expect users to use whitelist mode all the time to bypass interception in common Linux commands without intensive IO. Interception would be on in whitelist mode if the application is not in the bypass list.

The code of interception disabled are relatively much simpler. A few tests should be sufficient for testing.
If we do not force interception on with the new env, the interception is not enabled in many tests we have. Running more tests with interception is desired to discover bugs, so major tests are allocated for interception enabled.

> or are you saying you also want to add some tests to override the whitelist just to check those work even if performance is not good?
Yes. This is the case. OK. I will rename the env variable to D_IL_NO_WHITELIST. Thank you!

Reply:
Env variable is renamed. Comments are added. Thank you!